src/components/ProductsWithSearchInput/tests/ProductsWithSearchInput.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { vi } from 'vitest';
+
+import { productsMocks } from '@/mocks/products';
+import { render, screen, cleanup } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { getProducts } from '@/utils/api/getProducts';
+
+import ProductsWithSearchInput from '..';
+
+vi.mock(`@/utils/api/getProducts`);
+vi.mocked(getProducts).mockImplementation(async () => ({
+  products: productsMocks,
+}));
+
+describe(`Prices`, () => {
+  afterAll(() => {
+    vi.clearAllMocks();
+  });
+
+  afterEach(() => {
+    cleanup();
+    vi.resetAllMocks();
+  });
+
+  it(`When ProductsWithSearchInput is called and has products on products array, the products should be rendered`, async () => {
+    render(<ProductsWithSearchInput companyId="mockId" />);
+
+    expect(await screen.findByText(`Guaraná Antartica`)).toBeInTheDocument();
+  });
+
+  it(`When ProductsWithSearchInput is called and not have products on products array, the text "Não existem produtos cadastrados" should be rendered`, async () => {
+    vi.mocked(getProducts).mockImplementation(async () => ({
+      products: [],
+    }));
+
+    render(<ProductsWithSearchInput companyId="mockId" />);
+
+    expect(
+      await screen.findByText(`Não existem produtos cadastrados`)
+    ).toBeInTheDocument();
+
+    expect(screen.queryByText(`Guaraná Antartica`)).not.toBeInTheDocument();
+  });
+
```

Comment:
```suggestion
expect(screen.queryByText(`Guaraná Antartica`)).not.toBeInTheDocument();
```
i think that can be a fake positive, because you aren't await that to be in the document and the test should be resolved before render the product

Reply:
In this case not need await because I use QueryByText to verify if the element is not in the document

If I use wait findByText to check if an element is not in the document, the test case fails

Reply:
Is not fake positive

![Screen Shot 2023-08-31 at 8 31 12 AM](https://github.com/trukaSoftware/menu-digital/assets/65298393/28f551b8-e616-4177-873c-13b7979c2d8b)

Now I use await findByText and the tests broken

![Screen Shot 2023-08-31 at 8 32 50 AM](https://github.com/trukaSoftware/menu-digital/assets/65298393/919ef362-a035-4af7-b671-82e37dcab613)

Reply:
thats okay, nice!